From 489b7b36a129a1a6bf76d01af0feb1f1eee5fde5 Mon Sep 17 00:00:00 2001
From: Tommy Lin <lin.tommysc@inventec.com>
Date: Thu, 3 Mar 2022 01:17:17 +0000
Subject: [PATCH 1/3] Add a writable dbus interface MonitorSettings

- Add a writable dbus interface "MonitorSettings" for run-time configuring sensor.
---
 src/EntityManager.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/EntityManager.cpp b/src/EntityManager.cpp
index 8f29f92..f3642e8 100644
--- a/src/EntityManager.cpp
+++ b/src/EntityManager.cpp
@@ -59,8 +59,9 @@ const boost::container::flat_map<const char*, probe_type_codes, CmpStr>
                 {"FOUND", probe_type_codes::FOUND},
                 {"MATCH_ONE", probe_type_codes::MATCH_ONE}}};
 
-static constexpr std::array<const char*, 6> settableInterfaces = {
-    "FanProfile", "Pid", "Pid.Zone", "Stepwise", "Thresholds", "Polling"};
+static constexpr std::array<const char*, 7> settableInterfaces = {
+    "FanProfile", "Pid", "Pid.Zone", "Stepwise", "Thresholds", "Polling",
+    "MonitorSettings"};
 using JsonVariantType =
     std::variant<std::vector<std::string>, std::vector<double>, std::string,
                  int64_t, uint64_t, double, int32_t, uint32_t, int16_t,
-- 
2.33.0

